test/index.js

```diff
@@ -897,23 +1042,29 @@ describe('fs', () => {
   });
 
   it('ensurePathSync() - path is required', () => {
-    fs.ensurePathSync.should.to.throw('path is required!');
+    try {
+      fs.ensurePathSync();
+    } catch (err) {
+      err.message.should.eql('path is required!');
+    }
   });
 
-  it('ensureWriteStream()', () => {
+  it('ensureWriteStream()', async () => {
     const target = join(tmpDir, 'foo', 'bar.txt');
 
-    return fs.ensureWriteStream(target).then(stream => {
-      stream.path.should.eql(target);
+    const stream = await fs.ensureWriteStream(target);
+    stream.path.should.eql(target);
 
-      const streamPromise = new Promise((resolve, reject) => {
-        stream.on('error', reject);
-        stream.on('close', resolve);
-      });
+    const streamPromise = new Promise((resolve, reject) => {
```

Comment:
If the node version is 10 or higher, you can use the `Stream.finished` method to monitor stream completion and error events (friendly to the `util.promisify` method).